### PR TITLE
Module 4

### PR DESCRIPTION
**Pull request by Shaj-Amex**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,10 @@
 # DevOps Apprenticeship: Project Exercise
 
+## This is for scripting for Automating
```
Comment (Jonesey13) [inline README.md:3]: Where's the script (it's not listed in your description)?

File: setup.sh [hunk 2]
```diff
@@ -0,0 +1,31 @@
+#Step 1: sets the output if the script fails for error
+shebang | -ex
```
Comment (Jonesey13) [inline setup.sh]: You'll need a real shebang here:
```suggestion
#! /usr/bin/env bash -ex
```

File: setup.sh [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+#Step2: updates the list of available resources 
+sudo apt-get update -y
+
+#Step3: Install PyEnv Requisites
+sudo apt-get install -y build-essential libssl-dev zlib1g-dev libbz2-dev \
+libreadline-dev libsqlite3-dev wget curl llvm libncurses5-dev libncursesw5-dev \
+xz-utils tk-dev libffi-dev liblzma-dev python-openssl git
+
+#Step3: installs Python Version 
+sudo apt-get install python3-pip
```
Comment (Jonesey13) [inline setup.sh:12]: This installs pip (assuming you have python already installed)

File: setup.sh [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+#Step4: Find the Python Version
+python --version
+
+#Step5: installs poetry 
+curl -sSL https://raw.githubusercontent.com/python-poetry/
+poetry/master/get-poetry.py | python
```
Comment (Jonesey13) [inline setup.sh]: You'll should use a `\` to wrap this command across multiple lines:
```suggestion
#Step5: installs poetry 
curl -sSL https://raw.githubusercontent.com/python-poetry/ \
poetry/master/get-poetry.py | python
```

File: setup.sh [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+#Step6: Check for installed Poetry Version
+
+poetry --version
```
Comment (Jonesey13) [inline setup.sh]: Again this is redundant
```suggestion
#Step6: Check for installed Poetry Version

poetry --version
```
```suggestion
```

File: setup.sh [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+#Step3: Install PyEnv Requisites
+sudo apt-get install -y build-essential libssl-dev zlib1g-dev libbz2-dev \
+libreadline-dev libsqlite3-dev wget curl llvm libncurses5-dev libncursesw5-dev \
+xz-utils tk-dev libffi-dev liblzma-dev python-openssl git
+
+#Step3: installs Python Version 
+sudo apt-get install python3-pip
+
+#Step4: Find the Python Version
+python --version
```
Comment (Jonesey13) [inline setup.sh]: This isn't really needed to get things working
```suggestion
```

File: setup.sh [hunk 7]
```diff
@@ -0,0 +1,31 @@
+#Step 1: sets the output if the script fails for error
+shebang | -ex
+
+#Step2: updates the list of available resources 
+sudo apt-get update -y
+
+#Step3: Install PyEnv Requisites
+sudo apt-get install -y build-essential libssl-dev zlib1g-dev libbz2-dev \
+libreadline-dev libsqlite3-dev wget curl llvm libncurses5-dev libncursesw5-dev \
+xz-utils tk-dev libffi-dev liblzma-dev python-openssl git
```
Comment (Jonesey13) [inline setup.sh]: We're installing prerequisites for pyenv however we aren't actually installing pyenv itself (we still need git here however)
```suggestion
#Step3: Install git
sudo apt-get install -y git
```

**Review by Jonesey13 — CHANGES_REQUESTED** [review]
OK this generally looks quite good but there are some changes that need to be made to the setup script to make it usable (see my suggestions below).

File: README.md (hunk 1 above)
Comment (Shaj-Amex) [reply README.md:3]: This has been changed on Readme

File: setup.sh (hunk 2 above)
Comment (Shaj-Amex) [reply setup.sh]: Resolved by including the bash

File: setup.sh (hunk 3 above)
Comment (Shaj-Amex) [reply setup.sh:12]: Pip Installed

File: setup.sh (hunk 6 above)
Comment (Shaj-Amex) [reply setup.sh]: Removed the redundant Code

**Comment by Shaj-Amex** [comment]
Have modified the code based on review comments

File: README.md [hunk 8]
```diff
@@ -1,5 +1,10 @@
...
+ 
+The setup.sh file is script will error if there is an issue, setup Python, Poetry Install, Setup the dependencies. All you need is to copy TRELLO Credentials to .env for running the To Do App
```
Comment (Jonesey13) [inline README.md:5]: You might want to mention that this assumes python is already installed on the machine.

**Review by Jonesey13 — APPROVED** [review]
This generally looks OK (except it assumes a python installation which isn't mentioned in the README.md file).

Please feel free to merge this in whenever you are ready 👍

**Pull request merged**